### ISSUE TITLE
Order of Xaml code creates NullReferenceExeption

### ISSUE THREAD

**Issue by israelhuss** [head-first-csharp/fourth-edition#7]
On page 117 the book gives the Xaml code for the Label, TextBox, and TextBlock, but when running the code in that order you'll get a NullReferenceExeption, because when the TextChanged event is being raised when the TextBox is created, and at that point the TextBlock (named "number") is not created yet.
you just have to put the TextBlock (number) before the TextBox (numberTextBox), so when the event handler is created number is not Null.

**Comment by andrewstellman** [comment]
Hi @israelhuss. That's right! That's exactly how to solve the problem—which we do on the next page, in the "Sleuth it out" element. I really appreciate you letting us know about this, I'll add a "spoiler alert" to the bottom of that page so people don't get surprised and know that they'll learn about what's causing the NullReferenceException shortly.

**Comment by andrewstellman** [comment]
Oops, now I see what you mean! There's a page order error in the PDF—somehow pages 77 and 79 got swapped. I'll definitely make sure this gets fixed before we go to press.

**Comment by israelhuss** [comment]
Hi Andrew,
first of all, thank you so much for this book, it's amazing.
I keep on finding small typos and other problems, but I know that you
possibly already fixed them, my question is if you want to hear about them
or maybe its taken care of, or it's not possible to fix it anymore?
Thanx again
Israel

**Comment by andrewstellman** [comment]
Hi Israel,

I really appreciate that—thank you! Hopefully we've already taken care of those typos and small problems. (I'm doing a final pass through the code, and I found the same problem that you did—I should [hopefully] find all of the problems that keep the code from building.)

The version you're looking at is the QC1 version, so we did multiple rounds of fixing typos and small problems since then. It's possible that you caught something that we missed, but we don't really have a way for you to check that. I'm working on the last round of fixes this weekend—hopefully we'll find everything! 

I should have an updated PDF sometime next week that has the final version that we'll send to the printer. We'll definitely want to know about any typos that occur in that one. I'll make sure that PDF is updated on GItHub as soon as I have it available.

Thanks,
Andrew
